### PR TITLE
[16.0][MIG] account_invoice_line_default_account

### PR DESCRIPTION
**Pull request by rousseldenis**
- Use the standard compute method in order to get default account on new lines

- Avoid too much writings on partner by:
  - Grouping lines per invoice (as ... one partner per invoice)
  - taking the first candidate line and update the account.

- Split tests and add supplier invoices test cases

### CODE REVIEW

**Comment by rousseldenis** [comment]
@NL66278 @SirTakobi @eLBati @jbaudoux

**Comment by rousseldenis** [comment]
I'm wondering if we should avoid to write on partner field an account that is not 'expense' or 'income' ?

**Review by NL66278 — APPROVED** [review]
:+1:  LGTM

**Comment by rousseldenis** [comment]
> I'm wondering if we should avoid to write on partner field an account that is not 'expense' or 'income' ?

I've answered myself: the lambda domain does the job.

File: account_invoice_line_default_account/models/account_move_line.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        much writes, group lines per invoice and update with the first line
+        account
+        """
+        for invoice, lines in self.partition("move_id").items():
+            line_to_update = first(lines._get_updateable_income_expense_lines())
+            if not line_to_update:
+                continue
+            inv_type = invoice.move_type
+            if (
+                inv_type in ["in_invoice", "in_refund"]
```
Comment (jbaudoux) [inline account_invoice_line_default_account/models/account_move_line.py:74]:
```suggestion
                inv_type in ["in_invoice", "in_refund", "in_receipt"]
```

File: account_invoice_line_default_account/models/account_move_line.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            ):
+                if (
+                    line_to_update.account_id
+                    != invoice.partner_id.property_account_expense
+                ):
+                    invoice.partner_id.write(
+                        {"property_account_expense": line_to_update.account_id.id}
+                    )
+            elif (
+                inv_type in ["out_invoice", "out_refund"]
```
Comment (jbaudoux) [inline account_invoice_line_default_account/models/account_move_line.py:85]:
```suggestion
                inv_type in ["out_invoice", "out_refund", "out_receipt"]
```

**Review by jbaudoux — APPROVED** [review]
Looks good. Maybe you can extend the list of move type?
Not tested

**Review by Dranyel-Bosd — APPROVED** [review]
Functional Review, LGTM :+1: :sparkler:

**Review by bosd — APPROVED** [review]
LGTM

**Review by Dranyel-Bosd — APPROVED** [review]

**Review by eLBati — APPROVED** [review]
Tested. Thanks!

**Comment by andreampiovesana** [comment]
merge?

**Comment by ByteMeAsap** [comment]
Can we have this merged?

**Comment by rousseldenis** [comment]
@OCA/accounting-maintainers

**Comment by ByteMeAsap** [comment]
@pedrobaeza , can we have this merged?

**Pull request merged**